### PR TITLE
updating SC from golden, xocl not attached, baremetal

### PR DESCRIPTION
**Pull request by houlz0507**

### CODE REVIEW

File: src/runtime_src/core/pcie/linux/scan.cpp [hunk 1]
```diff
@@ -358,15 +358,20 @@ pcidev::pci_device::pci_device(const std::string& sysfs) : sysfs_name(sysfs)
...
+
     // Determine if the device is mgmt or user pf.
     std::string tmp;
     sysfs_get("", "mgmt_pf", err, tmp);
     if (err.empty()) {
         mgmt = true;
     } else {
+        mgmt = false;
         sysfs_get("", "user_pf", err, tmp);
         if (err.empty()) {
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/linux/scan.cpp]: empty clause of if statement?

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp [hunk 2]
```diff
@@ -142,45 +157,75 @@ static int updateSC(unsigned index, const char *file)
...
         return -EINVAL;
 
-    auto dev = pcidev::get_dev(index, true);
+    auto mgmt_dev = pcidev::get_dev(index, false);
+    std::shared_ptr<pcidev::pci_device> dev;
+    int i = 0;
+    for (dev = pcidev::get_dev(i, true); dev; dev = pcidev::get_dev(i, true)) {
+        if(dev->domain == mgmt_dev->domain &&
+            dev->bus == mgmt_dev->bus &&
+	    dev->dev == mgmt_dev->dev) {
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: pls fix indentation.

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp [hunk 3]
```diff
@@ -142,45 +157,75 @@ static int updateSC(unsigned index, const char *file)
...
+        if (shutdownStatus == 1){
+            /* Shutdown is done successfully. Returning from here */
+            break;
+        }
+        sleep(1);
     }
 
+    if (!shutdownStatus) {
+        std::cout << "Shutdown user function timeout." << std::endl;
+	return -ETIMEDOUT;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: indentation.

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp [hunk 4]
```diff
@@ -142,45 +157,75 @@ static int updateSC(unsigned index, const char *file)
...
-    std::shared_ptr<firmwareImage> bmc =
-        std::make_shared<firmwareImage>(file, BMC_FIRMWARE);
-    if (bmc->fail()) {
-        ret = -EINVAL;
-    } else {
-        ret = flasher.upgradeBMCFirmware(bmc.get());
+    dev->sysfs_put("", "remove", errmsg, "1\n");
+    if (!errmsg.empty()) {
+        std::cout << "Stopping user function failed" << std::endl;
+	return -EINVAL;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: ditto.

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp [hunk 5]
```diff
@@ -142,45 +157,75 @@ static int updateSC(unsigned index, const char *file)
...
     }
 
-    fs << "1\n";
-    fs.flush();
+    sleep(15);
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: sleep for 15 seconds?? why?

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp [hunk 6]
```diff
@@ -142,45 +157,75 @@ static int updateSC(unsigned index, const char *file)
...
+    ret = writeSCImage(flasher, file);
+
+    mgmt_dev->sysfs_put("", "dparent/rescan", errmsg, "1\n");
+    if (!errmsg.empty()) {
+        std::cout << "ERROR: " << errmsg << std::endl;
+    }
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: should we wait for user pf to show up again before return?

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp (hunk 5 above)
Comment (houlz0507) [reply src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: Upgrading SC immediately after pci hot reset will fail. And SC may not recover in this case. Waiting 15s might be safe. I have other simple steps to reproduce this and will file a p0 bug. 
I do not want to block the critical path at this time.

File: src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp (hunk 6 above)
Comment (houlz0507) [reply src/runtime_src/core/pcie/tools/xbmgmt/cmd_flash.cpp]: done

**Pull request merged**